### PR TITLE
7 feature 5 Import and Export Inventories

### PR DESCRIPTION
**Pull request by Robinrswanson**

### CODE REVIEW

File: src/main/java/interface_adaptors/gatewayReader.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    public gatewayReader(String filePath){
+        this.file = new File(filePath);
+    }
+
+    public gatewayReader(File file){
+        this.file = file;
+    }
+    public List<String[]> getData() throws IOException {
+        BufferedReader reader = new BufferedReader(new FileReader(file));
+        List<String[]> fileData = new ArrayList<>();
```
Comment (DaisyLi233) [inline src/main/java/interface_adaptors/gatewayReader.java]: I guess this fileData is the list of string that contains each line of the data in the file (which means each line of the data would stored as each element in fileData).

File: src/main/java/interface_adaptors/gatewayReader.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+public class gatewayReader {
+
+    private final File file;
+
+    public gatewayReader(String filePath){
+        this.file = new File(filePath);
+    }
+
+    public gatewayReader(File file){
+        this.file = file;
```
Comment (DaisyLi233) [inline src/main/java/interface_adaptors/gatewayReader.java]: I am not quite sure what this means. Are we supposed to assume we have a file type instance attribute stored in this class?

File: src/main/java/use_cases/Import.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import interface_adaptors.ImportPresenter;
+
+public class Import implements ImportInputBoundary{
+    private final ImportOutputBoundary presenter;
+    public Import(ImportOutputBoundary presenter){
+        this.presenter = presenter;
+    }
+    public String importDatabase(ImportDS importData){
+        List<String[]> data = importData.getImportData();
+        data.remove(0);
```
Comment (DaisyLi233) [inline src/main/java/use_cases/Import.java]: I assume this means removing the headline in our data file which does not contain any actual data.

File: src/main/java/interface_adaptors/ExportController.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    ExportInputBoundary exportUseCase;
+
+    public ExportController(ExportInputBoundary useCase){
+        this.exportUseCase = useCase;
+    }
+    public String export(){
+        ExportDS data = exportUseCase.extractDataStorage();
+        gatewayWriter writer = new gatewayWriter("src/main/java/database/Database.csv");
+        writer.rewriteFile(data.getDatabase());
+        return "src/main/java/database/Database.csv";
```
Comment (DaisyLi233) [inline src/main/java/interface_adaptors/ExportController.java]: It returns a string here so I guess it stores the csv file in this path. Does that mean we must have this file path in our laptop?

File: src/main/java/use_cases/ExportOutputBoundary.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package use_cases;
+
+import java.util.List;
+
+public interface ExportOutputBoundary {
+
+    public String prepareSuccess();
```
Comment (DaisyLi233) [inline src/main/java/use_cases/ExportOutputBoundary.java]: I think it makes sense we only have prepare success message here! We may not have any situation that does not export successfully.

**Review by DaisyLi233 — COMMENTED** [review]
I think this use case has mostly done! Import and Export features follow clean architecture structure pretty well and everything makes sense :).

**Comment by tonyxieuoft** [comment]
Everything looks good to me, from the import/export use case interactors/presenters/controllers and screens! 

It just occurred to me that we should probably use a builder to make each screen (something I'll implement when we're together); the elements and order of code for each is about the same.

**Review by wuhanyi70 — COMMENTED** [review]
Looks pretty good with a clear clean architecture frame. And the gateway reader and gateway writer could help with my history feature when I need to read the file or write files.

File: src/main/java/database/Database.csv [hunk 6]
```diff
@@ -0,0 +1,22 @@
+Apple,20.0,1000,[],Mon Nov 28 19:17:30 EST 2022,
```
Comment (BarackOllama578) [inline src/main/java/database/Database.csv:1]: You should consider documenting the format that the database would follow in a section in the readme while, and make reference to this in the classes that operate on the database

**Review by BarackOllama578 — APPROVED** [review]

**Pull request merged**